gibberish_server/controllers/room_controller.js

```diff
@@ -44,9 +44,12 @@ class RoomController {
         const roomId = req.body.roomId
         roomModel.getRoomInfoAsObject(roomId)
             .then(room => {
-                if(room.state != roomModel.STATE.GAME_WAITING) res.status(400).send('Game has started')
-                const index = room.players.findIndex(player => player.playerName == nickname)
-                if(index !== -1) {
+                console.log(room.players)
+                if (room.state != roomModel.STATE.GAME_WAITING) res.status(400).send('Game has started')
+                const index = room.players.findIndex(player => player.toJSON().playerName == nickname)
```

Comment:
why need to convert to json? can use player.name directly

Reply:
hmmm yeah i'm not sure, the POST /join_room test (the one to prevent duplicate nicknames)kept failing if i didn't put that, let me try again..

Reply:
yeah the test fails without the toJSON(), i'm not very sure why..

Reply:
i pass everything LOL